driver-core/src/main/java/com/datastax/driver/core/SimpleStatement.java

```diff
@@ -117,7 +114,24 @@ public String getQueryString() {
 
     @Override
     public ByteBuffer[] getValues() {
```

Comment:
> whenever computing hash for paging state

Mmm that's a bit unfortunate, it hadn't occurred to me when we introduced the paging state. That's not so bad because it's only on simple statements, and you can use `getPagingStateUnsafe` as a workaround.

But I think at some point we should retrofit `SimpleStatement` to implement `GettableByXxxData` and store encoded values. That's on a different branch and not related to this feature, so I'll create a separate ticket.